packages/react-component-library/src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react'
+import classNames from 'classnames'
+
+interface DrawerProps extends ComponentWithClass {
+  children?: React.ReactNode
+  onClose?: (event: React.FormEvent<HTMLButtonElement>) => void
+  isOpen?: boolean
+}
+
+export const Drawer: React.FC<DrawerProps> = ({
+  children,
+  onClose,
+  isOpen = false,
+}) => {
+  const [drawerOpen, setDrawerOpen] = useState(isOpen)
+
+  const classes = classNames('rn-drawer', {
+    'is-open': drawerOpen,
+  })
+
+  function handleClick(event: React.FormEvent<HTMLButtonElement>) {
+    setDrawerOpen(false)
+    onClose(event)
```

Comment:
Because `onClose` is nullable, this results in an error when not set, eg:

`Uncaught TypeError: onClose is not a function`